make_mesh.py

```diff
@@ -0,0 +1,256 @@
+	fin = numpy.copy(img_data)
+	fin[numpy.nonzero(img_data)] = numpy.fmax(img_data[numpy.nonzero(img_data)],origin[numpy.nonzero(img_data)])
+	return(fin,iso_surface);
+
+#Either take boundary from supplied mask or if not specified, from image directly
+def cut_img_mas(file_input,file_output,size,axis,direction,mask = None):
+	"""
+	Function to trim data matrix before mesh creation. Reads in nifti file and saves the trimmed image as nifti file.
+
+	Parameters:
+	-----------
+	file_input: str
+		file name of image to be loaded and cut (nifti format)
+	file_output: str
+		output file name
+	size : int
+		number of voxels to trim
+	axis : int
+		axis along which to trim (0,1,2)
+	direction : int
+		either trim form beginning af axis (0) inwards or from end of axis (1) inwards
+	mask : (X,Y,Z) array
+		optional. If given, boundary of image will be determined from the mask. Needed if image has boundary with non-zero entries
+
+	"""
+
+	path = os.path.abspath('.')
+	path = path + '/'
+	img= nibabel.load(path+file_input)
+	img_data = img.get_fdata()
+	header=img.header.copy()
+	if (mask != None):
+		mask= nibabel.load(mask)
+		mask_data = mask.get_fdata()
+		box = get_bounding_slices(mask_data)
+	else:
+		box = get_bounding_slices(img_data)
+	img_data = cut_img(img_data,box,size,axis,direction)
+	img_nifti=nibabel.Nifti1Image(img_data,None,header=header)
+	nibabel.save(img_nifti,file_output)
+	return
+
+#Define the boundin:g box of the data matrix. 
+def get_bounding_slices(img):
+	"""
+	Function to determine the boundaries of the given image.
+	
+	Parameters:
+	-----------
+	img : (X,Y,Z) array
+		Image data matrix of which boundaries are to be determined
+	
+	Returns:
+	--------
+	bbox : (Dim,2) array
+		array  with range of indices through the matrix that contain non-zero entries along each axis
+	
+	"""
+
+	dims = numpy.shape(img)
+	mask = img == 0
+	bbox = []
+	all_axis = numpy.arange(img.ndim)
+	for kdim in all_axis:
+		nk_dim = numpy.delete(all_axis, kdim)
+		mask_i = mask.all(axis=tuple(nk_dim))
+		dmask_i = numpy.diff(mask_i)
+		idx_i = numpy.nonzero(dmask_i)[0]
+		if len(idx_i) != 2:
+			#TODO: see if one boundary has been found, and check that)
+			print("No clear boundary found (no zero entries?) in dimension" + kdim)
+			print("Boundary of data matrix is returned instead")
+			idx_i = [0, dims[kdim]-2]
+		bbox.append([idx_i[0]+1, idx_i[1]+1])
+	return bbox
+
+# Trim image along specified axis, size input = voxel
+def cut_img(img,bbox,size,axis,direction):
+	"""
+	Function to trim an image data matrix.
+
+	Parameters:
+	-----------
+	img: (X,Y,Z) array
+		Image data matrix to be trimmed
+	bbox: (Dim,2) array
+		Integer values for each axis that specify bounding box of image as returend by get_bounding_slices()
+	size: int
+		number of voxels to trim
+	axis: int
+		axis along which to trim (0,1,2)
+	direction: int
```

Comment:
Why do we resolve this with an integer? `trim_starting_from` with either `"beginning"` and `"end"` string values would make it clearer.